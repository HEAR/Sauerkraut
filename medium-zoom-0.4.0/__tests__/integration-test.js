/* eslint-env jest */
import mediumZoom from '../src/medium-zoom'

global.requestAnimationFrame = cb => setTimeout(cb, 0)

const root = document.body

beforeEach(() => {
  while (root.firstChild) {
    root.removeChild(root.firstChild)
  }
})

describe('click', () => {
  test('on an image renders correctly', () => {
    const image = document.createElement('img')
    root.appendChild(image)

    mediumZoom('img')

    image.click()

    expect(image.className).toBe('medium-zoom-image')
    expect(document.querySelector('.medium-zoom-image--open')).toBeTruthy()
    expect(document.querySelector('.medium-zoom-overlay')).toBeTruthy()
    expect(root).toMatchSnapshot()
  })

  test('on a detached image doesnâ€™t add classes', () => {
    const image = document.createElement('img')
    root.appendChild(image)

    const zoom = mediumZoom('img')
    zoom.detach()

    image.click()

    expect(image.classList).toHaveLength(0)
    expect(root).toMatchSnapshot()
  })
})
