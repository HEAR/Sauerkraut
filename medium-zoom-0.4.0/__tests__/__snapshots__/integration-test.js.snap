// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click on a detached image doesnâ€™t add classes 1`] = `
<body>
  <img
    class=""
  />
</body>
`;

exports[`click on an image renders correctly 1`] = `
<body>
  <img
    class="medium-zoom-image"
    style="visibility: hidden;"
  />
  <div
    class="medium-zoom-overlay"
    style="background-color: rgb(255, 255, 255);"
  />
  <img
    class="medium-zoom-image medium-zoom-image--open"
    style="position: absolute; top: 0px; left: 0px; width: 0px; height: 0px;"
  />
</body>
`;
